src/shapes/textbox.class.js

```diff
@@ -92,15 +92,28 @@
         this._set('width', this.dynamicMinWidth);
       }
 
-      // calculate a styleMap that lets us know where styles as, as _textLines is separated by \n and wraps,
-      // but the style object line indices is by \n.
-      this._styleMap = this._generateStyleMap();
-
       // clear cache and re-calculate height
       this._clearCache();
       this.height = this._getTextHeight(ctx);
+      this._setLineWidths();
+    },
+
+    /**
+     * set the __lineWidths cache array to support
+     * functions that expect it to be filled
+     * @private
+     */
+    _setLineWidths: function() {
+      for (var i = 0, len = this._textLines.length; i < len; i++) {
+        this.__lineWidths[i] = this.width;
+      }
     },
 
+    /**
+     * Calculate a styleMap that lets us know where styles as, as _textLines is separated by \n and wraps,
+     * but the style object line indices is by \n.
```

Comment:
@asturur could you please rephrase this? it's difficult to understand it as it is